### DISCUSSION TITLE
How to Use Metal on Intel Mac?

### DISCUSSION THREAD

**Discussion by Spider-netizen** [abetlen/llama-cpp-python#1365] [Q&A]
Hi,

I'm trying to run a model using Metal on an intel Mac. I'm getting an error. This is the end of the traceback:

![image](https://github.com/abetlen/llama-cpp-python/assets/49787838/96a57c2e-a6df-47ea-8243-712b5e8edfcd)

Does anyone happen to know what I am missing?

**Comment by thiner** [comment]
I think "Metal" refers to M Series chip, which is ARM architecture, that's not built for Intel chip. You should try the CPU version, but that would be extremely slow in my opinion.

**Comment by andreped** [comment]
@thiner's answer is correct.

But for anyone wondering how to use Llama3 on Metal for ARM chip, you can install using:
```
CMAKE_ARGS="-DLLAMA_METAL=on" pip install llama-cpp-python
```

Then perform completion using this example (here I used the Q2 quantized Llama3 variant:
```
from llama_cpp import Llama

llm = Llama.from_pretrained(
    repo_id="QuantFactory/Meta-Llama-3-8B-Instruct-GGUF",
    filename="Meta-Llama-3-8B-Instruct.Q2_K.gguf",
    verbose=True,
    n_gpu_layers=-1,
)

output = llm(
      "Q: List the planet names in our solar system? A: ",
      max_tokens=32,
      stop=["Q:", "\n"],
      echo=True,
)
```

**Comment by Spider-netizen** [comment]
To my understanding Metal utilises mac's gpus for better performance. On
the official website it says it supports my computer. Also llama.cpp can be
build with Metal on my computer as it seems.

**Comment by thiner** [comment]
> To my understanding Metal utilises mac's gpus for better performance. 

Yes, it utilize Apple GPU, and Apple GPU is available only with M series chips.
